src/plugins/intel_gpu/src/graph/bucketize.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <string>
+
+#include "bucketize_inst.h"
+#include "intel_gpu/runtime/error_handler.hpp"
+#include "json_object.h"
+#include "primitive_type_base.h"
+
+namespace cldnn {
+primitive_type_id bucketize::type_id() {
+    static primitive_type_base<bucketize> instance;
+    return &instance;
+}
+
+layout bucketize_inst::calc_output_layout(bucketize_node const& node) {
+    assert(static_cast<bool>(node.get_primitive()->output_data_type) == false &&
+           "Output data type forcing is not supported for bucketize_node!");
+
+    auto desc = node.get_primitive();
+
+    auto input_layout = node.input(0).get_output_layout();
+    auto input_format = input_layout.format;
+
+    data_types output_type = desc->output_type;
+
+    if (input_layout.format.dimension() == 5) {
```

Comment:
As I understand, output shape of bucketize primitive is equal to the shape of first input. So you can just write something like:
```cpp
return layout{output_type, input_format, input_layout.size};
```